homeassistant/components/compit/__init__.py

```diff
@@ -0,0 +1,52 @@
+"""The Compit integration."""
+
+from __future__ import annotations
+
+import logging
+
+from compit_inext_api import CompitAPI, DeviceDefinitionsLoader
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import PLATFORMS
+from .coordinator import CompitDataUpdateCoordinator
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Compit from a config entry."""
+
+    session = async_get_clientsession(hass)
+    api = CompitAPI(entry.data["email"], entry.data["password"], session)
+    try:
+        system_info = await api.authenticate()
+
+        if system_info is False:
+            raise ConfigEntryAuthFailed(
+                f"Invalid credentials for {entry.data["email"]}"
+            )
+
+        device_definitions = await DeviceDefinitionsLoader.get_device_definitions(
+            hass.config.language
+        )
+
+        coordinator = CompitDataUpdateCoordinator(
+            hass, system_info.gates, api, device_definitions
+        )
+        await coordinator.async_config_entry_first_refresh()
+        entry.runtime_data = coordinator
+        await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    except ValueError as e:
+        _LOGGER.error("Value error: %s", e)
```

Comment:
It non-recoverable - for some reason we are not able to load device definitions. It shouldn't happen but get_device_definitions can throw this error when it can't load the definitions.